apps/E2E/src/CheckboxLegacy/pages/CheckboxLegacyPageObject.ts

```diff
@@ -5,61 +5,52 @@ import {
   HOMEPAGE_CHECKBOX_BUTTON,
   CHECKBOX_ON_PRESS,
 } from '../consts';
-import { BasePage, By } from '../../common/BasePage';
-
-/* This enum gives the spec file an EASY way to interact with SPECIFIC UI elements on the page.
- * The spec file should import this enum and use it when wanting to interact with different elements on the page. */
-export const enum CheckboxSelector {
-  Primary, //this._primaryComponent
-}
+import { BasePage, By, DesktopPlatform } from '../../common/BasePage';
+import { Attribute, AttributeValue } from '../../common/consts';
 
 class CheckboxLegacyPageObject extends BasePage {
   /******************************************************************/
   /**************** UI Element Interaction Methods ******************/
   /******************************************************************/
   async isCheckboxChecked(): Promise<boolean> {
-    return await (await this._primaryComponent).isSelected();
+    const checkbox = await this._primaryComponent;
+    if (this.platform === DesktopPlatform.Windows) {
+      // for native windows, .isSelected() always returns false. this is a workaround
+      return (await checkbox.getAttribute(Attribute.ToggleState)) === AttributeValue.on;
+    } else {
+      return await checkbox.isSelected();
+    }
   }
 
-  async waitForCheckboxChecked(timeout?: number): Promise<void> {
-    await browser.waitUntil(async () => await this.isCheckboxChecked(), {
-      timeout: timeout ?? this.waitForUiEvent,
-      timeoutMsg: 'The Checkbox was not toggled correctly',
-      interval: 1000,
-    });
+  /* Waits for the checkbox to be checked or unchecked if the new state is true or false. Returns true if the
+   * checkbox toggled to the new state. */
+  async waitForCheckboxToggle(newState: boolean, errorMessage?: string): Promise<boolean> {
```

Comment:
Remove optionality of `errorMessage` param.

Reply:
Done.